packages/sdk/src/core/schema/QueryParams.ts

```diff
@@ -1,19 +1,25 @@
+import { z } from "zod";
+
 /**
  * @internal
  */
 export const DEFAULT_QUERY_ALL_COUNT = 100;
 
+export const QueryAllParamsSchema = z
+  .object({
+    start: z.number().default(0),
+    count: z.number().default(DEFAULT_QUERY_ALL_COUNT),
+  })
+  .default({
+    start: 0,
+    count: DEFAULT_QUERY_ALL_COUNT,
+  });
+
 /**
  * Pagination Parameters
  * @public
  */
-export interface QueryAllParams {
-  /**
-   * the index to start from (default: 0)
-   */
-  start?: number;
-  /**
-   * how many items to return (default: 100)
-   */
-  count?: number;
-}
+export type QueryAllParams = Exclude<
```

Comment:
what are we excluding here?

Reply:
Excluding the possibility of undefined from the query params

Reply:
just don't `.default()` inside the schema object and you get the same thing

Reply:
Yea but the default is intentionally there